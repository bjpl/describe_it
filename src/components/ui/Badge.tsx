import React from "react";

interface BadgeProps {
  children: React.ReactNode;
  variant?: "default" | "secondary" | "outline" | "destructive";
  className?: string;
}

export const Badge: React.FC<BadgeProps> = ({
  children,
  variant = "default",
  className = "",
}) => {
  const variantStyles = {
    default: "bg-primary text-primary-foreground",
    secondary: "bg-secondary text-secondary-foreground",
    outline: "border-2 text-foreground",
    destructive: "bg-destructive text-destructive-foreground",
  };

  return (
    <span
      className={`inline-flex items-center rounded-full px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 ${variantStyles[variant]} ${className}`}
    >
      {children}
    </span>
  );
};

export default Badge;
