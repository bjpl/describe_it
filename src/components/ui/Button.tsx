import React from "react";

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?:
    | "default"
    | "secondary"
    | "destructive"
    | "outline"
    | "ghost"
    | "link";
  size?: "default" | "sm" | "lg" | "icon";
  children: React.ReactNode;
  loading?: boolean;
  leftIcon?: React.ReactNode;
  rightIcon?: React.ReactNode;
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      className = "",
      variant = "default",
      size = "default",
      children,
      loading = false,
      leftIcon,
      rightIcon,
      disabled,
      ...props
    },
    ref,
  ) => {
    const variantStyles = {
      default: "bg-primary text-primary-foreground hover:bg-primary/90",
      secondary: "bg-secondary text-secondary-foreground hover:bg-secondary/80",
      destructive:
        "bg-destructive text-destructive-foreground hover:bg-destructive/90",
      outline:
        "border border-input bg-background hover:bg-accent hover:text-accent-foreground",
      ghost: "hover:bg-accent hover:text-accent-foreground",
      link: "text-primary underline-offset-4 hover:underline",
    };

    const sizeStyles = {
      default: "h-10 px-4 py-2",
      sm: "h-9 rounded-md px-3",
      lg: "h-11 rounded-md px-8",
      icon: "h-10 w-10",
    };

    const isDisabled = disabled || loading;

    return (
      <button
        className={`inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ${variantStyles[variant]} ${sizeStyles[size]} ${className}`}
        ref={ref}
        disabled={isDisabled}
        aria-disabled={isDisabled}
        {...props}
      >
        {loading && (
          <svg
            className="animate-spin -ml-1 mr-2 h-4 w-4"
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            aria-hidden="true"
          >
            <circle
              className="opacity-25"
              cx="12"
              cy="12"
              r="10"
              stroke="currentColor"
              strokeWidth="4"
            />
            <path
              className="opacity-75"
              fill="currentColor"
              d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
            />
          </svg>
        )}
        {!loading && leftIcon && (
          <span className="mr-2" aria-hidden="true">
            {leftIcon}
          </span>
        )}
        {children}
        {!loading && rightIcon && (
          <span className="ml-2" aria-hidden="true">
            {rightIcon}
          </span>
        )}
      </button>
    );
  },
);

Button.displayName = "Button";

export default Button;
