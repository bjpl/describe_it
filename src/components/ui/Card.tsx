import React from "react";

export interface CardProps {
  children: React.ReactNode;
  className?: string;
  clickable?: boolean;
  onClick?: () => void;
  role?: string;
  'aria-label'?: string;
  'aria-labelledby'?: string;
}

export interface CardHeaderProps {
  children: React.ReactNode;
  className?: string;
}

export interface CardTitleProps {
  children: React.ReactNode;
  className?: string;
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
}

export interface CardDescriptionProps {
  children: React.ReactNode;
  className?: string;
}

export interface CardContentProps {
  children: React.ReactNode;
  className?: string;
}

export interface CardFooterProps {
  children: React.ReactNode;
  className?: string;
}

export const Card: React.FC<CardProps> = ({ 
  children, 
  className = "", 
  clickable = false,
  onClick,
  role,
  ...props 
}) => {
  const Component = clickable ? 'button' : 'div';
  
  return (
    <Component
      className={`rounded-lg border bg-card text-card-foreground shadow-sm ${
        clickable ? 'hover:shadow-md transition-shadow cursor-pointer focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2' : ''
      } ${className}`}
      onClick={clickable ? onClick : undefined}
      role={role}
      {...props}
    >
      {children}
    </Component>
  );
};

export const CardHeader: React.FC<CardHeaderProps> = ({
  children,
  className = "",
}) => {
  return (
    <div className={`flex flex-col space-y-1.5 p-6 ${className}`}>
      {children}
    </div>
  );
};

export const CardTitle: React.FC<CardTitleProps> = ({
  children,
  className = "",
  as: Component = 'h3',
}) => {
  return (
    <Component
      className={`text-2xl font-semibold leading-none tracking-tight ${className}`}
    >
      {children}
    </Component>
  );
};

export const CardDescription: React.FC<CardDescriptionProps> = ({
  children,
  className = "",
}) => {
  return (
    <p className={`text-sm text-muted-foreground ${className}`}>{children}</p>
  );
};

export const CardContent: React.FC<CardContentProps> = ({
  children,
  className = "",
}) => {
  return <div className={`p-6 pt-0 ${className}`}>{children}</div>;
};

export const CardFooter: React.FC<CardFooterProps> = ({
  children,
  className = "",
}) => {
  return (
    <div className={`flex items-center p-6 pt-0 ${className}`}>{children}</div>
  );
};
