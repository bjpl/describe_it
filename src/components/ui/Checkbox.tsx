"use client";

import React from "react";
import { cn } from "@/lib/utils";
import { Check, Minus } from "lucide-react";

export interface CheckboxProps {
  checked?: boolean;
  indeterminate?: boolean;
  onChange?: (checked: boolean) => void;
  onClick?: (event: React.MouseEvent) => void;
  disabled?: boolean;
  className?: string;
  size?: "sm" | "md" | "lg";
  "aria-label"?: string;
  "aria-labelledby"?: string;
  "aria-describedby"?: string;
}

export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
  (
    {
      checked = false,
      indeterminate = false,
      onChange,
      onClick,
      disabled = false,
      className,
      size = "md",
      ...props
    },
    ref
  ) => {
    const sizeClasses = {
      sm: "h-4 w-4",
      md: "h-5 w-5",
      lg: "h-6 w-6",
    };

    const iconSizeClasses = {
      sm: "h-3 w-3",
      md: "h-4 w-4",
      lg: "h-5 w-5",
    };

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      onChange?.(event.target.checked);
    };

    const handleClick = (event: React.MouseEvent) => {
      onClick?.(event);
    };

    return (
      <div className="relative inline-flex items-center">
        <input
          ref={ref}
          type="checkbox"
          checked={checked}
          onChange={handleChange}
          onClick={handleClick}
          disabled={disabled}
          className="sr-only"
          {...props}
        />
        <div
          className={cn(
            "relative flex items-center justify-center border-2 rounded transition-all duration-200",
            sizeClasses[size],
            "focus-within:ring-2 focus-within:ring-ring focus-within:ring-offset-2",
            checked || indeterminate
              ? "bg-primary border-primary text-primary-foreground"
              : "border-input bg-background hover:border-primary/50",
            disabled && "opacity-50 cursor-not-allowed",
            !disabled && "cursor-pointer",
            className
          )}
          role="checkbox"
          aria-checked={indeterminate ? "mixed" : checked}
          tabIndex={disabled ? -1 : 0}
          onKeyDown={(e) => {
            if ((e.key === "Space" || e.key === "Enter") && !disabled) {
              e.preventDefault();
              onChange?.(!checked);
            }
          }}
        >
          {(checked || indeterminate) && (
            <div className="flex items-center justify-center">
              {indeterminate ? (
                <Minus className={cn(iconSizeClasses[size], "stroke-[3]")} />
              ) : (
                <Check className={cn(iconSizeClasses[size], "stroke-[3]")} />
              )}
            </div>
          )}
        </div>
      </div>
    );
  }
);

Checkbox.displayName = "Checkbox";

export default Checkbox;
