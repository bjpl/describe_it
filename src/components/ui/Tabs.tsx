import React, { useState, createContext, useContext } from 'react';

interface TabsContextValue {
  value: string;
  onValueChange: (value: string) => void;
}

const TabsContext = createContext<TabsContextValue | undefined>(undefined);

interface TabsProps {
  defaultValue?: string;
  value?: string;
  onValueChange?: (value: string) => void;
  className?: string;
  children: React.ReactNode;
}

export const Tabs: React.FC<TabsProps> = ({ 
  defaultValue = '', 
  value: controlledValue,
  onValueChange,
  className = '',
  children 
}) => {
  const [internalValue, setInternalValue] = useState(defaultValue);
  const value = controlledValue ?? internalValue;
  
  const handleValueChange = (newValue: string) => {
    if (onValueChange) {
      onValueChange(newValue);
    } else {
      setInternalValue(newValue);
    }
  };

  return (
    <TabsContext.Provider value={{ value, onValueChange: handleValueChange }}>
      <div className={`w-full ${className}`}>
        {children}
      </div>
    </TabsContext.Provider>
  );
};

interface TabsListProps {
  className?: string;
  children: React.ReactNode;
}

export const TabsList: React.FC<TabsListProps> = ({ className = '', children }) => {
  return (
    <div className={`inline-flex h-10 items-center justify-center rounded-md bg-muted p-1 text-muted-foreground ${className}`}>
      {children}
    </div>
  );
};

interface TabsTriggerProps {
  value: string;
  className?: string;
  children: React.ReactNode;
  disabled?: boolean;
}

export const TabsTrigger: React.FC<TabsTriggerProps> = ({ 
  value, 
  className = '', 
  children,
  disabled = false 
}) => {
  const context = useContext(TabsContext);
  if (!context) throw new Error('TabsTrigger must be used within Tabs');
  
  const { value: activeValue, onValueChange } = context;
  const isActive = activeValue === value;

  return (
    <button
      onClick={() => !disabled && onValueChange(value)}
      disabled={disabled}
      className={`inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ${
        isActive 
          ? 'bg-background text-foreground shadow-sm' 
          : 'hover:bg-accent hover:text-accent-foreground'
      } ${className}`}
    >
      {children}
    </button>
  );
};

interface TabsContentProps {
  value: string;
  className?: string;
  children: React.ReactNode;
}

export const TabsContent: React.FC<TabsContentProps> = ({ 
  value, 
  className = '', 
  children 
}) => {
  const context = useContext(TabsContext);
  if (!context) throw new Error('TabsContent must be used within Tabs');
  
  const { value: activeValue } = context;
  
  if (activeValue !== value) return null;

  return (
    <div 
      className={`mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 ${className}`}
    >
      {children}
    </div>
  );
};

// Export named components for compatibility
export const Root = Tabs;
export const List = TabsList;
export const Trigger = TabsTrigger;
export const Content = TabsContent;