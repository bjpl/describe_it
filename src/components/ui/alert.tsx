/**
 * Alert UI Component
 * Displays important messages and notifications
 */

import React from 'react';
import { cn } from '@/lib/utils';
import { cva, type VariantProps } from 'class-variance-authority';

const alertVariants = cva(
  'relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950',
  {
    variants: {
      variant: {
        default: 'bg-white text-gray-950 border-gray-200',
        destructive:
          'border-red-200 text-red-800 bg-red-50 [&>svg]:text-red-600',
        warning:
          'border-yellow-200 text-yellow-800 bg-yellow-50 [&>svg]:text-yellow-600',
        success:
          'border-green-200 text-green-800 bg-green-50 [&>svg]:text-green-600',
        info:
          'border-blue-200 text-blue-800 bg-blue-50 [&>svg]:text-blue-600',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  }
);

interface AlertProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof alertVariants> {}

const Alert = React.forwardRef<HTMLDivElement, AlertProps>(
  ({ className, variant, ...props }, ref) => (
    <div
      ref={ref}
      role="alert"
      className={cn(alertVariants({ variant }), className)}
      {...props}
    />
  )
);
Alert.displayName = 'Alert';

interface AlertTitleProps extends React.HTMLAttributes<HTMLHeadingElement> {}

const AlertTitle = React.forwardRef<HTMLParagraphElement, AlertTitleProps>(
  ({ className, ...props }, ref) => (
    <h5
      ref={ref}
      className={cn('mb-1 font-medium leading-none tracking-tight', className)}
      {...props}
    />
  )
);
AlertTitle.displayName = 'AlertTitle';

interface AlertDescriptionProps
  extends React.HTMLAttributes<HTMLParagraphElement> {}

const AlertDescription = React.forwardRef<
  HTMLParagraphElement,
  AlertDescriptionProps
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn('text-sm [&_p]:leading-relaxed', className)}
    {...props}
  />
));
AlertDescription.displayName = 'AlertDescription';

export { Alert, AlertTitle, AlertDescription };