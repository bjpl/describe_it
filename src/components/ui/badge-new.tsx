/**
 * Badge Component - Custom implementation to replace missing Radix UI component
 */
import React from 'react';
import { cn } from '@/lib/utils';

interface BadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
  variant?: 'default' | 'secondary' | 'success' | 'warning' | 'error' | 'outline';
  size?: 'sm' | 'md' | 'lg';
}

const Badge: React.FC<BadgeProps> = ({ 
  className, 
  variant = 'default', 
  size = 'md',
  children,
  ...props 
}) => {
  const baseClasses = "inline-flex items-center font-semibold rounded-full border transition-colors";
  
  const variantClasses = {
    default: "bg-blue-100 text-blue-800 border-blue-200 dark:bg-blue-900 dark:text-blue-200",
    secondary: "bg-gray-100 text-gray-800 border-gray-200 dark:bg-gray-800 dark:text-gray-200",
    success: "bg-green-100 text-green-800 border-green-200 dark:bg-green-900 dark:text-green-200",
    warning: "bg-yellow-100 text-yellow-800 border-yellow-200 dark:bg-yellow-900 dark:text-yellow-200",
    error: "bg-red-100 text-red-800 border-red-200 dark:bg-red-900 dark:text-red-200",
    outline: "bg-transparent text-gray-800 border-gray-300 dark:text-gray-200 dark:border-gray-600",
  };
  
  const sizeClasses = {
    sm: "px-2 py-0.5 text-xs",
    md: "px-2.5 py-1 text-sm", 
    lg: "px-3 py-1.5 text-base",
  };

  return (
    <span
      className={cn(
        baseClasses,
        variantClasses[variant],
        sizeClasses[size],
        className
      )}
      {...props}
    >
      {children}
    </span>
  );
};

export { Badge };
export type { BadgeProps };