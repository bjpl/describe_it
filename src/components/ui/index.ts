// Core UI Components
export { Button } from "./Button";
export { Card, CardHeader, CardTitle, CardDescription, CardContent, CardFooter } from "./Card";
export { Input } from "./Input";
export { Progress } from "./Progress";
export { Badge } from "./Badge";
export { RadioGroup } from "./RadioGroup";
export { Select } from "./Select";
export { Tabs } from "./Tabs";

// Enhanced Components
export {
  Skeleton,
  EnhancedSkeleton,
  SkeletonGroup,
  TextSkeleton,
  ProfileSkeleton,
  CardSkeletonEnhanced,
  TableSkeleton,
} from "./Skeleton";

export type { Toast } from "./Toast";
export {
  ToastProvider,
  useToast,
  toast,
} from "./Toast";

export {
  Modal,
  ModalHeader,
  ModalTitle,
  ModalDescription,
  ModalContent,
  ModalFooter,
  ConfirmModal,
} from "./Modal";

export {
  Dropdown,
  SimpleDropdown,
} from "./Dropdown";

export {
  Tooltip,
  Popover,
} from "./Tooltip";

export {
  DataTable,
} from "./DataTable";

export {
  Checkbox,
} from "./Checkbox";

export {
  FormField,
  FormLabel,
  FormDescription,
  FormMessage,
  FormInput,
  FormTextarea,
  useFormValidation,
} from "./FormField";

// Motion Components - Export individual components from MotionComponents
export {
  MotionDiv,
  MotionButton,
  MotionSpan,
  MotionP,
  MotionH1,
  MotionH2,
  MotionH3,
  MotionH4,
  MotionH5,
  MotionH6,
  MotionForm,
  MotionInput,
  MotionTextarea,
  MotionSelect,
  MotionHeader,
  MotionSection,
  MotionArticle,
  MotionAside,
  MotionNav,
  MotionMain,
  MotionFooter,
  MotionA,
  MotionImg,
  MotionUl,
  MotionOl,
  MotionLi,
  MotionLink,
} from "./MotionComponents";

// Fallback components (if needed for compatibility)
export {
  MotionDiv as FallbackMotionDiv,
  MotionButton as FallbackMotionButton,
  MotionSpan as FallbackMotionSpan,
  FallbackAnimatePresence,
} from "./MotionFallback";

// Types - Export from their respective source files
export type { ButtonProps } from "./Button";
export type {
  CardProps,
} from "./Card";
export type { ModalProps } from "./Modal";
export type { TooltipProps, PopoverProps } from "./Tooltip";
export type { DropdownProps, DropdownItem } from "./Dropdown";
export type { DataTableProps, Column } from "./DataTable";
export type { CheckboxProps } from "./Checkbox";
export type {
  FormFieldProps,
  FormLabelProps,
  FormMessageProps,
  FormInputProps,
  FormTextareaProps,
  ValidationRule,
  UseFormValidationProps,
} from "./FormField";
