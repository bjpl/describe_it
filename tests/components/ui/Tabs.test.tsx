import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import '@testing-library/jest-dom';
import { Tabs, TabsList, TabsTrigger, TabsContent } from '@/components/ui/Tabs';

describe('Tabs Components', () => {
  describe('Tabs Container', () => {
    it('should render children', () => {
      render(
        <Tabs>
          <div>Tab content</div>
        </Tabs>
      );
      expect(screen.getByText('Tab content')).toBeInTheDocument();
    });

    it('should apply custom className', () => {
      const { container } = render(
        <Tabs className="custom-tabs">
          <div>Content</div>
        </Tabs>
      );
      expect(container.firstChild).toHaveClass('custom-tabs');
    });

    it('should use defaultValue for initial state', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsContent value="tab1">Tab 1 Content</TabsContent>
          <TabsContent value="tab2">Tab 2 Content</TabsContent>
        </Tabs>
      );
      expect(screen.getByText('Tab 1 Content')).toBeInTheDocument();
      expect(screen.queryByText('Tab 2 Content')).not.toBeInTheDocument();
    });
  });

  describe('Controlled vs Uncontrolled', () => {
    it('should work in uncontrolled mode', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
            <TabsTrigger value="tab2">Tab 2</TabsTrigger>
          </TabsList>
          <TabsContent value="tab1">Content 1</TabsContent>
          <TabsContent value="tab2">Content 2</TabsContent>
        </Tabs>
      );

      expect(screen.getByText('Content 1')).toBeInTheDocument();

      fireEvent.click(screen.getByText('Tab 2'));
      expect(screen.getByText('Content 2')).toBeInTheDocument();
      expect(screen.queryByText('Content 1')).not.toBeInTheDocument();
    });

    it('should work in controlled mode', () => {
      const handleChange = vi.fn();
      const { rerender } = render(
        <Tabs value="tab1" onValueChange={handleChange}>
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
            <TabsTrigger value="tab2">Tab 2</TabsTrigger>
          </TabsList>
          <TabsContent value="tab1">Content 1</TabsContent>
          <TabsContent value="tab2">Content 2</TabsContent>
        </Tabs>
      );

      expect(screen.getByText('Content 1')).toBeInTheDocument();

      fireEvent.click(screen.getByText('Tab 2'));
      expect(handleChange).toHaveBeenCalledWith('tab2');

      // Simulate parent updating value
      rerender(
        <Tabs value="tab2" onValueChange={handleChange}>
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
            <TabsTrigger value="tab2">Tab 2</TabsTrigger>
          </TabsList>
          <TabsContent value="tab1">Content 1</TabsContent>
          <TabsContent value="tab2">Content 2</TabsContent>
        </Tabs>
      );

      expect(screen.getByText('Content 2')).toBeInTheDocument();
    });
  });

  describe('TabsList Component', () => {
    it('should render children', () => {
      render(
        <TabsList>
          <button>Tab 1</button>
        </TabsList>
      );
      expect(screen.getByText('Tab 1')).toBeInTheDocument();
    });

    it('should have default list styles', () => {
      const { container } = render(
        <TabsList>
          <button>Tab</button>
        </TabsList>
      );
      const list = container.firstChild;
      expect(list).toHaveClass('inline-flex', 'h-10', 'rounded-md', 'bg-muted');
    });

    it('should apply custom className', () => {
      const { container } = render(
        <TabsList className="custom-list">
          <button>Tab</button>
        </TabsList>
      );
      expect(container.firstChild).toHaveClass('custom-list');
    });
  });

  describe('TabsTrigger Component', () => {
    it('should render trigger button', () => {
      render(
        <Tabs>
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
          </TabsList>
        </Tabs>
      );
      expect(screen.getByText('Tab 1')).toBeInTheDocument();
    });

    it('should throw error when used outside Tabs context', () => {
      // Suppress console.error for this test
      const consoleSpy = vi.spyOn(console, 'error').mockImplementation(() => {});

      expect(() => {
        render(<TabsTrigger value="tab1">Tab 1</TabsTrigger>);
      }).toThrow('TabsTrigger must be used within Tabs');

      consoleSpy.mockRestore();
    });

    it('should apply active styles when selected', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsList>
            <TabsTrigger value="tab1">Active Tab</TabsTrigger>
            <TabsTrigger value="tab2">Inactive Tab</TabsTrigger>
          </TabsList>
        </Tabs>
      );

      const activeTab = screen.getByText('Active Tab');
      expect(activeTab).toHaveClass('bg-background', 'text-foreground', 'shadow-sm');
    });

    it('should apply inactive styles when not selected', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsList>
            <TabsTrigger value="tab1">Active Tab</TabsTrigger>
            <TabsTrigger value="tab2">Inactive Tab</TabsTrigger>
          </TabsList>
        </Tabs>
      );

      const inactiveTab = screen.getByText('Inactive Tab');
      expect(inactiveTab).toHaveClass('hover:bg-accent');
    });

    it('should change active tab on click', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
            <TabsTrigger value="tab2">Tab 2</TabsTrigger>
          </TabsList>
          <TabsContent value="tab1">Content 1</TabsContent>
          <TabsContent value="tab2">Content 2</TabsContent>
        </Tabs>
      );

      fireEvent.click(screen.getByText('Tab 2'));
      expect(screen.getByText('Content 2')).toBeInTheDocument();
    });

    it('should be disabled when disabled prop is true', () => {
      render(
        <Tabs>
          <TabsList>
            <TabsTrigger value="tab1" disabled>Disabled Tab</TabsTrigger>
          </TabsList>
        </Tabs>
      );

      const disabledTab = screen.getByText('Disabled Tab');
      expect(disabledTab).toBeDisabled();
    });

    it('should not change tab when disabled', () => {
      const handleChange = vi.fn();
      render(
        <Tabs onValueChange={handleChange}>
          <TabsList>
            <TabsTrigger value="tab1" disabled>Disabled Tab</TabsTrigger>
          </TabsList>
        </Tabs>
      );

      fireEvent.click(screen.getByText('Disabled Tab'));
      expect(handleChange).not.toHaveBeenCalled();
    });

    it('should apply custom className', () => {
      render(
        <Tabs>
          <TabsList>
            <TabsTrigger value="tab1" className="custom-trigger">Tab 1</TabsTrigger>
          </TabsList>
        </Tabs>
      );

      expect(screen.getByText('Tab 1')).toHaveClass('custom-trigger');
    });

    it('should have focus styles', () => {
      render(
        <Tabs>
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
          </TabsList>
        </Tabs>
      );

      expect(screen.getByText('Tab 1')).toHaveClass('focus-visible:outline-none', 'focus-visible:ring-2');
    });
  });

  describe('TabsContent Component', () => {
    it('should render content when value matches', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsContent value="tab1">Tab 1 Content</TabsContent>
        </Tabs>
      );
      expect(screen.getByText('Tab 1 Content')).toBeInTheDocument();
    });

    it('should not render content when value does not match', () => {
      render(
        <Tabs defaultValue="tab1">
          <TabsContent value="tab2">Tab 2 Content</TabsContent>
        </Tabs>
      );
      expect(screen.queryByText('Tab 2 Content')).not.toBeInTheDocument();
    });

    it('should throw error when used outside Tabs context', () => {
      const consoleSpy = vi.spyOn(console, 'error').mockImplementation(() => {});

      expect(() => {
        render(<TabsContent value="tab1">Content</TabsContent>);
      }).toThrow('TabsContent must be used within Tabs');

      consoleSpy.mockRestore();
    });

    it('should apply custom className', () => {
      const { container } = render(
        <Tabs defaultValue="tab1">
          <TabsContent value="tab1" className="custom-content">Content</TabsContent>
        </Tabs>
      );

      const content = container.querySelector('.custom-content');
      expect(content).toBeInTheDocument();
    });

    it('should have focus ring styles', () => {
      const { container } = render(
        <Tabs defaultValue="tab1">
          <TabsContent value="tab1">Content</TabsContent>
        </Tabs>
      );

      const content = screen.getByText('Content').parentElement;
      expect(content).toHaveClass('focus-visible:outline-none', 'focus-visible:ring-2');
    });
  });

  describe('Complete Tabs Example', () => {
    it('should render complete tabs with multiple panels', () => {
      render(
        <Tabs defaultValue="overview">
          <TabsList>
            <TabsTrigger value="overview">Overview</TabsTrigger>
            <TabsTrigger value="details">Details</TabsTrigger>
            <TabsTrigger value="settings">Settings</TabsTrigger>
          </TabsList>
          <TabsContent value="overview">Overview content</TabsContent>
          <TabsContent value="details">Details content</TabsContent>
          <TabsContent value="settings">Settings content</TabsContent>
        </Tabs>
      );

      expect(screen.getByText('Overview content')).toBeInTheDocument();

      fireEvent.click(screen.getByText('Details'));
      expect(screen.getByText('Details content')).toBeInTheDocument();
      expect(screen.queryByText('Overview content')).not.toBeInTheDocument();

      fireEvent.click(screen.getByText('Settings'));
      expect(screen.getByText('Settings content')).toBeInTheDocument();
      expect(screen.queryByText('Details content')).not.toBeInTheDocument();
    });
  });

  describe('Snapshot Tests', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Tabs defaultValue="tab1">
          <TabsList>
            <TabsTrigger value="tab1">Tab 1</TabsTrigger>
            <TabsTrigger value="tab2">Tab 2</TabsTrigger>
          </TabsList>
          <TabsContent value="tab1">Content 1</TabsContent>
          <TabsContent value="tab2">Content 2</TabsContent>
        </Tabs>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
