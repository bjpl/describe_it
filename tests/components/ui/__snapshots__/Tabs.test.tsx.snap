// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs Components > Snapshot Tests > should match snapshot 1`] = `
<div
  class="w-full "
>
  <div
    class="inline-flex h-10 items-center justify-center rounded-md bg-muted p-1 text-muted-foreground "
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-background text-foreground shadow-sm "
    >
      Tab 1
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground "
    >
      Tab 2
    </button>
  </div>
  <div
    class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 "
  >
    Content 1
  </div>
</div>
`;
